volume/volume_propagation_linux_test.go

```diff
@@ -0,0 +1,49 @@
+// +build linux
+
+package volume
+
+import (
+	"strings"
+	"testing"
+)
+
+func TestParseMountSpecPropagation(t *testing.T) {
+	var (
+		valid   []string
+		invalid map[string]string
+	)
+
+	valid = []string{
+		"/hostPath:/containerPath:shared",
+		"/hostPath:/containerPath:slave",
+		"/hostPath:/containerPath:private",
+		"/hostPath:/containerPath:ro,shared",
+		"/hostPath:/containerPath:ro,slave",
+		"/hostPath:/containerPath:ro,private",
+		"/hostPath:/containerPath:ro,z,shared",
+		"/hostPath:/containerPath:ro,Z,slave",
```

Comment:
Do the mount options have to appear in the specified order?
Would `Z,slave,ro` and `slave,ro,Z` be legal syntax as well?
I'd like to see test cases that clarify this.
This is relevant for the shell completions.